### PR TITLE
Fix races around touching state.Group outside of the event loop

### PR DESCRIPTION
**Pull request by mstoykov**
There is no locking around the `state.Group` and it is modified by calls to `group()` while on the event loop. So for the time being to modify it - it is needed to be on the event loop.

Also the adding of the tag isn't needed as that is done automatically by `group()`

### CODE REVIEW

**Comment by mstoykov** [comment]
found while testing for #542

File: common/frame_session.go
Comment (ankur22) [inline common/frame_session.go:522]: I'm assuming this wouldn't be an issue if whatever action that was used was running on the event loop i.e. was async?

Comment (inancgumus) [inline common/frame_session.go:522]: Should we remove the comment?

```suggestion
```

File: common/frame_session.go
Comment (inancgumus) [inline common/frame_session.go:697]: Should we remove the comment?

```suggestion
```

File: common/frame_session.go
Comment (mstoykov) [reply common/frame_session.go:522]: yes if it was *on* the event loop it would've been fine as a `group()` call couldn't have changed the value ... as it also needs to run on the event loop.

> was async? 

async is way too generic term ... in practice it *is* async - it doesn't run syncrhonisly ;) It also doesn't run on the event loop - which is in general the idea of async ... anything. To free the event loop to do other stuff and as they don't run on the event loop (and consequently on/by the JS VM) you can do multiple stuff in parallel (or at least concurrently). The problem is that IMO "async" is used too generically. 

From what I could gather, this is just something that *always* ran asynchronously. And maybe the addition of *async* APIs made it more likely to race ... or maybe it always was possible to race - I haven't tested with a really old version of xk6-browser and `-race` :shrug: 

But yes - if we can make this code ot run on the event loop - that will make it possible to keep the lines, which is why I just commented them. Unfortunately that (IMO) will require that most if not all of the APIs are async as otherwise this code will not run if the JS VM is currently blocked on doing an *synchronous* call.

I also am not certain that this (or some of the surrounding) code does not *unblock* a certain syncrhnous call that will otherwise not continue. 

Not to speak that making this run on the event loop will be *way* bigger change, which is why I opted for dropping some info so I get rid of this race condition instead of doing something way more involved.

Comment (ankur22) [reply common/frame_session.go:522]: I'm happy with this approach for now.

Comment (ankur22) [reply common/frame_session.go:522]: I'd prefer to have them left commented out and maybe even link to a issue so that we know why it has been commented out and what it will take to uncomment them.

Comment (inancgumus) [reply common/frame_session.go:522]: > if we can make this code ot run on the event loop - that will make it possible to keep the lines, which is why I just commented them

👍

**Review by ankur22 — APPROVED** [review]

**Review by inancgumus — APPROVED** [review]
LGTM, but I believe we need a new issue to add `Group` support back in the xk6-browser.

**Comment by mstoykov** [comment]
>  a new issue to add Group support back in the xk6-browser.

what do you mean?


the tag `group` is still added - the code that is removed here didn't need to exist. 

Do you want an issue to make the logging of `group` to be brought back for the two cases where it was removed?

**Comment by inancgumus** [comment]
> Do you want an issue to make the logging of `group` to be brought back for the two cases where it was removed?

Never mind, I'm confused. It's good as it is 👍

**Pull request merged**
